### PR TITLE
WFS updates

### PR DESCRIPTION
**Pull request by delroth**
Just to get a Windows build...

### CODE REVIEW

File: Source/Core/Common/FileUtil.cpp [hunk 1]
```diff
@@ -600,10 +593,12 @@ void CopyDir(const std::string& source_path, const std::string& dest_path)
     {
       if (!Exists(dest))
         File::CreateFullPath(dest + DIR_SEP);
-      CopyDir(source, dest);
+      CopyDir(source, dest, destructive);
     }
-    else if (!Exists(dest))
+    else if (!Exists(dest) && !destructive)
```
Comment (BhaaLseN) [inline Source/Core/Common/FileUtil.cpp]: Some braces would be nice here, since the `if` has some.

File: Source/Core/Core/IOS/WFS/WFSI.cpp [hunk 2]
```diff
@@ -21,6 +21,20 @@
 #include "Core/IOS/WFS/WFSSRV.h"
 #include "DiscIO/NANDContentLoader.h"
 
+namespace
+{
+std::string TitleIdStr(u64 tid)
```
Comment (BhaaLseN) [inline Source/Core/Core/IOS/WFS/WFSI.cpp]: About time! 👍

File: Source/Core/Core/IOS/WFS/WFSSRV.cpp [hunk 3]
```diff
@@ -246,6 +365,19 @@ IPCCommandResult WFSSRV::IOCtl(const IOCtlRequest& request)
...
+s32 WFSSRV::Rename(const std::string& source, const std::string& dest) const
+{
+  const bool opened = std::any_of(m_fds.begin(), m_fds.end(), [&](const auto& fd) {
+    return fd.in_use && fd.path == source;
+  });
+
+  if (opened)
+    return WFS_FILE_IS_OPENED;
+
+  File::Rename(WFS::NativePath(source), WFS::NativePath(dest));
```
Comment (BhaaLseN) [inline Source/Core/Core/IOS/WFS/WFSSRV.cpp]: Can a rename fail (and should it return something else instead of "success")?

Comment (leoetlino) [inline Source/Core/Core/IOS/WFS/WFSSRV.cpp]: Errors are pretty unlikely to happen (read/write failures aren't checked either), so I didn't bother checking for failure here. I guess we could handle the most common failure case (ENOENT) though?

**Comment by JMC47** [comment]
LGTM, still works for DQX

**Comment by Ebola16** [comment]
This PR caused [Major slowdown due to "WFS updates"](https://bugs.dolphin-emu.org/issues/10499)

**Pull request merged**